src/contextProviders/ConfirmationDialogProvider/index.tsx

```diff
@@ -14,34 +35,34 @@ export function ConfirmationDialogContextWrapper({ children }: any) {
     const [state, setState] = useState({
         confirmationDialogQuestion: "",
         confirmationDialogAction: (e: React.MouseEvent) => {},
+        cancelConfirmDialogAction: (e: React.MouseEvent) => {},
         leavePage: false,
         isError: false,
     });
-    const cancelConfirmation = () => setCnfirmationDialogOpen(false);
-
-    const showConfirmDialog = (
-        question: string,
-        action: (e: React.MouseEvent) => void,
-        leavePage: boolean = false,
-        isError: boolean = false
-    ) => {
+    const cancelConfirmDialog = () => setCnfirmationDialogOpen(false);
+
+    const showConfirmDialog = ({ question, confirmAction, cancelAction, leavePage, isError }: ShowConfirmDialog) => {
         setCnfirmationDialogOpen(true);
         setState({
             confirmationDialogQuestion: question,
-            leavePage,
-            isError,
+            leavePage: !!leavePage,
+            isError: !!isError,
             confirmationDialogAction: (e: React.MouseEvent) => {
-                cancelConfirmation();
-                action(e);
+                cancelConfirmDialog();
+                confirmAction(e);
+            },
+            cancelConfirmDialogAction: (e: React.MouseEvent) => {
+                cancelConfirmDialog();
+                if (cancelAction) cancelAction(e);
             },
```

Comment:
I don't oversee the entire implementation, but could it be reduced to 1 dialogaction perhaps? 

```ts
            confirmationDialogAction: (e: React.MouseEvent) => {
                cancelConfirmDialog();  // Might be better to just call this closeConfirmDialog?
                if (cancelAction) {
                    cancelAction(e);
                } else {
                    confirmAction(e);
                }
            },
 
```

Because in my opinion having both a `cancelConfirmDialogAction` as well as a `confirmationDialogAction` (which, to be consistent, should be `confirmConfirmationDialogAction`  :P ) seems a bit unnecessary when it could just be a boolean